### PR TITLE
feat(model): user model, service, controller, and router

### PR DESCRIPTION
**Pull request by alanwu4321**
### What
Create `User` model, service, controller, and router

### How
- Create Moongse Model
- Create UsersService to interact with the model object
- Create a Controller that utilizes UsersService
- Create UserRouter and specified Controller handlers for each endpoint
- Expose UserRouter in the main root Router

### TODO
- [ ] Complete user interface
- [ ] Update API yml

### Demo 
POST endpoint to create users (with reviews) and GET endpoint to see all users
image
image

### CODE REVIEW

**Comment by 17yongga** [comment]
Looks good

File: server/api/models/post.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  {
+    id: { type: Number, unique: true },
+    // TODO
+  },
+  {
+    collection: "posts",
+  }
+);
+
+posts.plugin(AutoIncrement, { inc_field: "id" });
```
Comment (17yongga) [inline server/api/models/post.ts:20]: Would this need to be post_id?

**Review by 17yongga — CHANGES_REQUESTED** [review]

**Comment by ryanhow10** [comment]
Looks good, maybe adding the schema in a README would be beneficial for the rest of the team.

File: server/api/models/post.ts (hunk 1 above)
Comment (alanwu4321) [reply server/api/models/post.ts:20]: Good idea! I should make it consistent with `user_id` and `review_id` as well

**Comment by alanwu4321** [comment]
> Looks good, maybe adding the schema in a README would be beneficial for the rest of the team.

I will start a schema section in README to document Mongoose schema changes

**Review by 17yongga — APPROVED** [review]
Nice, all good

**Comment by alanwu4321** [comment]
Automatically add `created_at` and `updated_at` to User and Review schema

**Pull request merged**